Builds/CMake/soci_patch.cmake

```diff
@@ -1,15 +1,17 @@
 # This patches unsigned-types.h in the soci official sources
 # so as to remove type range check exceptions that cause
 # us trouble when using boost::optional to select int values
-file (STRINGS include/soci/unsigned-types.h sourcecode)
-foreach (line_ ${sourcecode})
-  if (line_ MATCHES "^[ \\t]+throw[ ]+soci_error[ ]*\\([ ]*\"Value outside of allowed.+$")
-    set (line_ "//${CMAKE_MATCH_0}")
-  endif ()
-  file (APPEND include/soci/unsigned-types.h.patched "${line_}\n")
-endforeach ()
-file (RENAME include/soci/unsigned-types.h include/soci/unsigned-types.h.orig)
-file (RENAME include/soci/unsigned-types.h.patched include/soci/unsigned-types.h)
+if(NOT EXISTS "include/soci/unsigned-types.h.orig")
```

Comment:
Today I learned that re-running the patch step is a [known and recently-fixed bug in CMake](https://gitlab.kitware.com/cmake/cmake/-/issues/21086). I think the better workaround is to just `return()` if the `.orig` file is seen. Nothing in `soci_patch.cmake` needs to run on subsequent patch steps. Please also include a link to the CMake bug report to explain the early return.

Reply:
Can you please fast-forward merge the branch [fix-cmake-soci-patch](https://github.com/thejohnfreeman/rippled/tree/fix-cmake-soci-patch) from my fork?